pandas/core/ops.py

```diff
@@ -1197,8 +1200,35 @@ def wrapper(self, other, axis=None):
         if is_datetime64_dtype(self) or is_datetime64tz_dtype(self):
             # Dispatch to DatetimeIndex to ensure identical
             # Series/Index behavior
+            if (isinstance(other, datetime.date) and
+                    not isinstance(other, datetime.datetime)):
```

Comment:
is this check needed? (I would not expect it ever to be true if it is a datetime.date ?

Reply:
Apparently datetime.datetime subclasses datetime.date

https://bugs.python.org/issue28878

Reply:
rather than inlineing this, I think would be better to move to a module level function, maybe
``other = maybe_coerce_other(other, op)``

Reply:
You mean just the new bit I added, right?

I'm inclined to leave it here since we'll just be deleting it soon (is 0.24.0 too soon?).

Reply:
> Apparently datetime.datetime subclasses datetime.date

Ah, only checked the other way around :-)

Yeah, if this is only temporary code I would just leave it here

Reply:
ok, is there an issue to fix this for 0.24.0? (can you add a TODO mention here)